### PR TITLE
Add Dockerfile and build scripts

### PR DESCRIPTION
**Pull request by erickmendonca**
This adds a way to run the application using docker containers. It's useful to deploy this bot on environments that are friendly to containerized applications.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,6 +16,9 @@
     "build": "tsc -p tsconfig.json",
     "build:watch": "tsc -p tsconfig.json --watch",
     "dev": "nodemon --exec \"npm start\"",
+    "docker:build": "docker build -t probot-auto-merge-alpine .",
```
Comment (bobvanderlinden) [inline package.json]: This doesn't align with what is defined in the readme.

```suggestion
    "docker:build": "docker build -t probot-auto-merge .",
```

File: package.json [hunk 2]
```diff
@@ -16,6 +16,9 @@
...
+    "docker:run": "docker run -p 0.0.0.0:3000:3000 probot-auto-merge-alpine",
```
Comment (bobvanderlinden) [inline package.json]:
```suggestion
    "docker:run": "docker run -p 0.0.0.0:3000:3000 probot-auto-merge",
```

File: package.json (hunk 1 above)
Comment (erickmendonca) [reply package.json]: Oh, thanks for the catch!

**Pull request merged**